app/models/concerns/team_duplication.rb

```
@@ -7,215 +7,109 @@ module TeamDuplication
+        team.save(validate: false)
+        self.store_clones(team)
+        return team
       end
     end
 
-    def self.log_error(e, t)
-      self.notify_error(e, { team_id: t.id }, RequestStore[:request])
-      Rails.logger.error "[Team Duplication] Could not duplicate team #{t.slug}: #{e.message} #{e.backtrace.join("\n")}"
-    end
-
-    def self.flag_project_medias(original, copy)
-      original.team.is_being_copied = true if original.is_a?(ProjectMedia)
-      copy.team.is_being_copied = true if copy.is_a?(ProjectMedia)
+    def self.alter_team_copy(copy)
+      copy.name = @custom_name || "Copy of #{copy.name}"
+      copy.slug = @custom_slug || copy.generate_copy_slug
+      copy.is_being_copied = true
+      copy.set_slack_notifications_enabled = false
     end
 
-    def self.set_mapping(object, copy)
-      key = object.class_name.to_sym
-      @mapping[key] ||= {}
-      @mapping[key][object.id] = copy
+    def self.alter_project_copy(copy)
+      copy.generate_token(true)
+      copy.set_slack_notifications_enabled = false
     end
 
-    def self.copy_image(original, copy)
-      [:logo, :lead_image, :file].each do |image|
-        next unless original.respond_to?(image) && original.respond_to?("#{image}=") && original.send(image)
-        [original.send(image)].flatten.each do |img|
-          copy.send("#{image}=", open(img.url)) if img.url =~ /^#{CONFIG['storage']['endpoint']}/
-        end
+    def self.alter_copy_by_type(original, copy)
+      if original.is_a?(Team)
+        self.alter_team_copy(copy)
+      elsif original.is_a?(Project)
+        self.alter_project_copy(copy)
+      elsif original.is_a?(TagText)
+        copy.team_id = @team_id if !@team_id.nil?
       end
-    end
-
-    def self.versions_log_mapping(original, copy)
-      if original.respond_to?(:get_versions_log)
-        original.get_versions_log.find_each do |log|
-          self.set_mapping(log, copy)
-        end
+      copy.save(validate: false)
```

Please replace this by `copy.save!` - we want to raise an exception if the copy is not valid